rocAL/rocAL/source/loaders/audio/audio_read_and_decode.cpp

```diff
@@ -0,0 +1,159 @@
+    _actual_decoded_samples.resize(_batch_size);
+    _actual_decoded_channels.resize(_batch_size);
+    _original_channels.resize(_batch_size);
+    _original_samples.resize(_batch_size);
+    _original_sample_rates.resize(_batch_size);
+    _decoder_config = decoder_config;
+    if ((_decoder_config._type != DecoderType::SKIP_DECODE)) {
+        for (int i = 0; i < batch_size; i++) {
+            _decoder[i] = create_audio_decoder(decoder_config);
+        }
+    }
+    _reader = create_reader(reader_config);
+    _input_path = reader_config.path();
+    if(_input_path.back() != '/')
+        _input_path = _input_path + "/";
+}
+
+void
+AudioReadAndDecode::reset() {
+    // TODO: Reload audios from the folder if needed
+    _reader->reset();
+}
+
+size_t
+AudioReadAndDecode::count() {
+    return _reader->count_items();
+}
+
+size_t
+AudioReadAndDecode::last_batch_padded_size() {
+    return _reader->last_batch_padded_size();
+}
+
+LoaderModuleStatus
+AudioReadAndDecode::load(float* buff,
+                         std::vector<std::string>& names,
+                         const size_t max_decoded_samples,
+                         const size_t max_decoded_channels,
+                         std::vector<uint32_t> &roi_samples,
+                         std::vector<uint32_t> &roi_channels,
+                         std::vector<uint32_t> &actual_samples,
+                         std::vector<uint32_t> &actual_channels,
+                         std::vector<float> &actual_sample_rates) {
+    if(max_decoded_samples == 0 || max_decoded_channels == 0 )
+        THROW("Zero audio dimension is not valid")
+    if(!buff)
+        THROW("Null pointer passed as output buffer")
+    if(_reader->count_items() < _batch_size)
+        return LoaderModuleStatus::NO_MORE_DATA_TO_READ;
+    // load audios/frames from the disk and push them as a large audio onto the buff
+    unsigned file_counter = 0;
+    const size_t audio_size = max_decoded_samples * max_decoded_channels;
+    // Decode with the channels and size equal to a single audio
+    // File read is done serially since I/O parallelization does not work very well.
+    _file_load_time.start();// Debug timing
+    while ((file_counter != _batch_size) && _reader->count_items() > 0) {
+        size_t fsize = _reader->open();
+        if (fsize == 0) {
+            WRN("Opened file " + _reader->id() + " of size 0");
+            continue;
+        }
+        _audio_names[file_counter] = _reader->id();
+        _audio_file_path[file_counter] = _reader->file_path();
+        _reader->close();
+        file_counter++;
+    }
+    _file_load_time.end();// Debug timing
+    _decode_time.start();// Debug timing
+    if (_decoder_config._type != DecoderType::SKIP_DECODE) {
+        for (size_t i = 0; i < _batch_size; i++){
+            _decompressed_buff_ptrs[i] = buff + (audio_size * i);
+        }
+#pragma omp parallel for num_threads(8)  // default(none) TBD: option disabled in Ubuntu 20.04
```

Comment:
This must be changed to copu_threads_count like in image

Reply:
We can do this in next PR?

Reply:
@swetha097 Can we address this in this PR itself. 
How many files are required for this?

Reply:
Okay Shobana, 5 file changes - additional files were required